src/Compilers/Core/Portable/PEWriter/SigningUtilities.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Security.Cryptography;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Emit;
+
+namespace Microsoft.Cci
+{
+    internal static class SigningUtilities
+    {
+        internal static byte[] CalculateRsaSignature(IEnumerable<Blob> content, RSAParameters privateKey)
+        {
+            var hash = CalculateSha1(content);
+
+            using (var rsa = RSA.Create())
+            {
+                rsa.ImportParameters(privateKey);
+                var signature = rsa.SignHash(hash, HashAlgorithmName.SHA1, RSASignaturePadding.Pkcs1);
+                Array.Reverse(signature);
+                return signature;
+            }
+        }
+
+        private static byte[] CalculateSha1(IEnumerable<Blob> content)
+        {
+            using (var hash = IncrementalHash.CreateHash(HashAlgorithmName.SHA1))
+            {
+                foreach (var blob in content)
+                {
+                    var segment = blob.GetBytes();
+                    hash.AppendData(segment.Array, segment.Offset, segment.Count);
+                }
```

Comment:
There's an existing extension method, `IncrementalHashExtensions.AppendData(IncrementalHash, IEnumerable<Blob>)`, for the loop.

Reply:
Oh neat!  Thanks!